crates/bevy_reflect/src/path/access.rs

```diff
@@ -0,0 +1,240 @@
+use std::fmt;
+
+use super::ReflectPathError;
+use crate::{Reflect, ReflectMut, ReflectRef, VariantType};
+use thiserror::Error;
+
+type InnerResult<T> = Result<Option<T>, Error<'static>>;
+
+#[derive(Debug, PartialEq, Eq, Error)]
+pub(super) enum Error<'a> {
+    #[error(
+        "the current {ty} doesn't have the {} {}",
+        access.kind(),
+        access.display_value(),
+    )]
+    Access { ty: Type, access: AccessRef<'a> },
+
+    #[error("invalid type shape: expected {expected} but found a reflect {actual}")]
```

Comment:
I think the wording here could be a bit clearer. Maybe we can use the `-like` suffix to make it clear that we're expecting a certain shape and not that actual type (i.e. `struct-like` works for both structs and struct enum variants).
```suggestion
    #[error("invalid type shape: expected {expected}-like but found {actual}")]
```

Reply:
yeah, but `Type` here is not really a "type" but a "type shape", The error message would read as follow: "expected struct but found tuple". I don't think there is such a thing as a "struct-like" it's either a struct or not a struct.

I'll see if I can rename Type to TypeShape without hurting readability.